src/index.ts

```diff
@@ -222,23 +222,31 @@ function encodeNumber(buffer: Uint8Array, i: number, number: Numeric): number {
   return i
 }
 
-export class StringTable extends Array {
+export class StringTable {
+  strings = new Array<string>()
   #encodings = new Map<string, Uint8Array>()
   #positions = new Map<string, number>()
 
+  constructor(values?: Array<string>) {
+    if (values === undefined) {
+      // Create a StringTable and populate first elemnt with ''
+      this.dedup('')
+    } else {
+      // Need to copy over manually to ensure the lookup map is correct
+      // Should probably check that first element is the empty string
```

Comment:
I agree that we should check for empty string is present, but I am not sure that adding it silently if not present is the good thing to do: if the user has built its own table, forgetting about the empty string, then if we add it, we will also make all the indices off by one.
I would be more in favour of just bailing out and returning an error, but this is where my limited JS knowledge ends ;)

Reply:
All string table id references have a default value of zero, which maps to the empty string, so it makes sense to me to just ensure the empty case is there. People shouldn't be using array indexes as the id anyway rather than using the dedup method.

Reply:
I agree that people should use the `dedup` method, but the API offers the possibility to the client to pass its own  string table as a string array, and thus should check that the table makes sense.
I think we should either:
* force the client to use `StringTable` and forbid passing a string array in `ProfileInput`
* allow a string array but check that it is correct (at least that it has the empty string as first element) and fail otherwise
WDYT ?

Reply:
That functionality is not used or documented though and only existed because StringTable _was_ an array before so it could be used with the `push` helper which would end up building an array rather than a StringTable as the input. Could just make it only accept a StringTable.

Reply:
I failed to understand the conclusion. Should we just remove the array passing API ?

Reply:
Yeah, I would just make it _only_ accept a StringTable. Much easier to be certain about the correctness of the data that way.